pymc_bart/pgbart.py

```diff
@@ -76,7 +76,7 @@ def __init__(
         self.bart = model.values_to_rvs[value_bart].owner.op
 
         if isinstance(self.bart.X, TensorSharedVariable):
-            self.X = self.bart.X.eval()
+            self.X = self.bart.X.get_value(borrow=False)
         else:
             self.X = self.bart.X
```

Comment:
What is the type in this case? `TensorConstant`? `np.array`? I was thinking that it may be cast to float if it's a shared variable with integer type originally or something like that.

Reply:
The else branch is expecting a numpy array. It is ok if integer as cast to floats.

Reply:
Then I suggest you assert that. Because it could very well be a `at.cast(at.shared(X))` (if you coerced it to `floatX` before) or `at.constant(X)` (if it was given as `ConstantData`), and so on.